tools/sim/bridge.py

```diff
@@ -80,138 +84,164 @@ def go(q):
+      vehicle = world.try_spawn_actor(vehicle_bp, spawn_point)
+
+    if args.autopilot:
+      vehicle.set_autopilot(True)
+    
+    print(vehicle.get_speed_limit())
+    if vehicle is not None:
+      print("\n\n -- > vehicle has been configured!!!")
+
+    # make tires less slippery
+    wheel_control = carla.WheelPhysicsControl(tire_friction=5)
+    physics_control = vehicle.get_physics_control()
+    physics_control.mass = 1326
+    physics_control.wheels = [wheel_control]*4
+    physics_control.torque_curve = [[20.0, 500.0], [5000.0, 500.0]]
+    physics_control.gear_switch_time = 0.0
+    vehicle.apply_physics_control(physics_control)
+
+    blueprint = blueprint_library.find('sensor.camera.rgb')
+    blueprint.set_attribute('image_size_x', str(W))
+    blueprint.set_attribute('image_size_y', str(H))
+    blueprint.set_attribute('fov', '70')
+    blueprint.set_attribute('sensor_tick', '0.05')
+    #transform = carla.Transform(carla.Location(x=0.8, z=1.45))
+
+    transform = [carla.Transform(carla.Location(x=-5.5, z=2.8), carla.Rotation(pitch=-15)),
+    carla.Transform(carla.Location(x=1.6, z=1.7))]
+
+    camera = world.spawn_actor(blueprint, transform[0], attach_to=vehicle)
+    camera.listen(cam_callback)
+  
+    # reenable IMU
+    imu_bp = blueprint_library.find('sensor.other.imu')
+    imu = world.spawn_actor(imu_bp, transform[0], attach_to=vehicle)
+    
+    imu.listen(imu_callback)
+
+    # can loop
+    sendcan = messaging.sub_sock('sendcan')
+    rk = Ratekeeper(100, print_delay_threshold=0.05)
+
+    # init
+    A_throttle = 2.
+    A_brake = 2.
+    A_steer_torque = 1.
+    fake_wheel = FakeSteeringWheel()
+    is_openpilot_engaged = False
+    in_reverse = False
+
+    throttle_out = 0
+    brake_out = 1
+    steer_angle_out = 0
+
+    vc = carla.VehicleControl(throttle=throttle_out, 
+                              steer=steer_angle_out / 3.14, 
+                              brake=brake_out, 
+                              reverse=in_reverse)
+
+    print ("vehicle control created!!!")
+
+    while 1:
+      cruise_button = 0
+
+      # check for a input message, this will not block
+      if not q.empty():
+        message = q.get()
+
+        m = message.split('_')
+        if m[0] == "steer":
+          steer_angle_out = float(m[1])
+          fake_wheel.set_angle(steer_angle_out)  # touching the wheel overrides fake wheel angle
+          # print(" === steering overriden === ")
+        if m[0] == "throttle":
+          throttle_out = float(m[1]) / 100.
+          if throttle_out > 0.3:
+            cruise_button = CruiseButtons.CANCEL
+            is_openpilot_engaged = False
+        if m[0] == "brake":
+          brake_out = float(m[1]) / 100.
+          if brake_out > 0.3:
+            cruise_button = CruiseButtons.CANCEL
+            is_openpilot_engaged = False
+        if m[0] == "reverse":
+          in_reverse = not in_reverse
```

Comment:
seems that VehicleControl is not picking up this?

Reply:
If you mean... reverse is not being handle by the keyboard_ctrl... you are right
![image](https://user-images.githubusercontent.com/15021357/88297771-b631b500-ccd6-11ea-9183-a816bedc044e.png)
only cruise up/down/cancel are being capture